ca_tools/scripts/pos_controller_pid.py

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env python
+
+import rospy
+import math
+from nav_msgs.msg import Odometry
+from geometry_msgs.msg import Twist
+from tf.transformations import euler_from_quaternion
+
+
+
+class Execute_Draw_Square:
+
+    # Time variables
+    now=float
+    timeChange=float
+    lastTime=0.0
+
+    # Angle variables
+    Current_angle=0.0
+    Setpoint_angle=0.0
+    Input_angle=0.0
+    error_angle=0.0
+    errSum_angle=0.0
+    Max_output_angle=0.0
+    Min_output_angle =0.0
+
+    # Lenght variables
+    Current_length=0.0
+    Setpoint_length=0.0
+    Input_length=0.0
+    error_length=0.0
+    errSum_length=0.0
+    Max_output_length=0.0
+    Min_output_length=0.0
+
+    # PID variables
+    kp=float
+    ki=float
+    kd=float
+
+    #State Variables
+    Setpoint_variable=bool
+    state="FIRST"
+
+    #Twist Variable
+    twist = Twist()
+
+    #Odometry Flag
+    Odom_flag=False
+
+    def __init__(self):
+
+        # PID variables
+        self.kp=1
+        self.ki=0.5
+        self.kd=0.4
+
+        # Set State Variable
+        self.Set_variable=True
+        self.state="FIRST"
+
+        self.Max_output_angle = 0.4
+        self.Min_output_angle = -0.4
+
+        self.Max_output_length = 0.4
+        self.Min_output_length = -0.4
+
+    def callback(self,data):
+        odom_pos = data
+        q = (odom_pos.pose.pose.orientation.x,
+             odom_pos.pose.pose.orientation.y,
+             odom_pos.pose.pose.orientation.z,
+             odom_pos.pose.pose.orientation.w)
+        euler_angles = euler_from_quaternion(q)
```

Comment:
You can do this:
```Python
euler_angles = euler_from_quaternion(q)
self.Current_angle = euler_angles[2]
```
in one line using this:
```Python
_, _, self.Current_angle = euler_from_quaternion(q)
```

Reply:
Also, names of the variables are wrong because you're mixing CamelCase with snake_case.